idea/src/org/jetbrains/kotlin/idea/inspections/coroutines/DeferredResultUnusedInspection.kt

```diff
@@ -22,10 +23,12 @@ class DeferredResultUnusedInspection(@JvmField var standardOnly: Boolean = false
                 (!inspection.standardOnly || expression.calleeExpression?.text in shortNames),
     callChecker = fun(resolvedCall, inspection): Boolean {
         if (inspection !is DeferredResultUnusedInspection) return false
+        val resultingDescriptor = resolvedCall.resultingDescriptor
+        if (resultingDescriptor.fqNameSafe == FqName("kotlin.test.assertNotNull")) return false
```

Comment:
But the called function may have an arbitrary name, see https://youtrack.jetbrains.com/issue/KT-33741#focus=streamItem-27-3697460.0-0